File: app/src/main/java/develop/y/zhzl/search/presenter/SearchPresenterImpl.java
package develop.y.zhzl.search.presenter;

import android.text.TextUtils;

import com.rxnetwork.bus.RxBus;
import com.rxnetwork.bus.RxBusCallBack;
import com.rxnetwork.manager.RxNetWork;

import java.util.List;

import develop.y.zhzl.list.model.ListModel;
import develop.y.zhzl.search.view.SearchView;
import framework.api.Api;
import framework.data.Constant;
import framework.mvp.PresenterCompat;
import framework.sql.SearchSuffixDb;
import rx.Subscription;

/**
 * by y on 2016/8/7.
 */
public class SearchPresenterImpl extends PresenterCompat<List<ListModel>, SearchView>
        implements SearchPresenter, RxBusCallBack<String> {

    private String suffix = "";
    private final Subscription annalSubscription;

    public SearchPresenterImpl(SearchView mView) {
        super(mView);
        annalSubscription = RxBus.getInstance().toSubscription(Constant.ANNAL_TAG, String.class, this);
    }

    @Override
    protected void onNetWorkNext(List<ListModel> data) {
        getView().hideExplanation();
        getView().adapterRemove();
        getView().rxNetWorkSuccess(data);
        SearchSuffixDb.insert(suffix);
    }


    @Override
    public void netWorkRequest(final String suffix, String limit) {
        this.suffix = suffix;
        getView().hideExplanation();
        if (TextUtils.isEmpty(suffix)) {
            getView().showExplanation();
            getView().suffixIsEmpty();
            return;
        }
        int limits;
        if (TextUtils.isEmpty(limit)) {
            limits = Constant.LIMIT;
        } else {
            limits = Integer.valueOf(limit);
        }
        RxNetWork.
                getInstance().
                getApi(RxNetWork.observable(Api.ZLService.class).getList(suffix, limits, Constant.OFFSET), this);
    }

    @Override
    public void onNext(String data) {
        netWorkRequest(data, "");
//        RxBus.getInstance().unregister(Constant.ANNAL_TAG, annalSubscription);
    }

    @Override
    public void onError(Throwable throwable) {

    }
}

File: app/src/main/java/develop/y/zhzl/search/presenter/SearchPresenterImpl.kt
package develop.y.zhzl.search.presenter

import android.text.TextUtils

import com.rxnetwork.bus.RxBus
import com.rxnetwork.bus.RxBusCallBack
import com.rxnetwork.manager.RxNetWork

import develop.y.zhzl.list.model.ListModel
import develop.y.zhzl.search.view.SearchView
import framework.api.Api
import framework.data.Constant
import framework.mvp.PresenterCompat
import framework.sql.sqlutil.SearchSuffixDb
import rx.Subscription

/**
 * by y on 2016/8/7.
 */
class SearchPresenterImpl(mView: SearchView) : PresenterCompat<List<ListModel>, SearchView>(mView), SearchPresenter, RxBusCallBack<String> {

    private var suffix = ""
    private val annalSubscription: Subscription = RxBus.getInstance().toSubscription(Constant.ANNAL_TAG, String::class.java, this)

    override fun onNetWorkNext(data: List<ListModel>) {
        view.hideExplanation()
        view.adapterRemove()
        view.rxNetWorkSuccess(data)
        SearchSuffixDb.insert(suffix)
    }


    override fun netWorkRequest(suffix: String, limit: String) {
        this.suffix = suffix
        view.hideExplanation()
        if (TextUtils.isEmpty(suffix)) {
            view.showExplanation()
            view.suffixIsEmpty()
            return
        }
        val limits: Int
        if (TextUtils.isEmpty(limit)) {
            limits = Constant.LIMIT
        } else {
            limits = Integer.valueOf(limit)!!
        }
        RxNetWork.getInstance().getApi(RxNetWork.observable(Api.ZLService::class.java).getList(suffix, limits, Constant.OFFSET), this)
    }

    override fun onNext(data: String) {
        netWorkRequest(data, "")
        //        RxBus.getInstance().unregister(Constant.ANNAL_TAG, annalSubscription);
    }

    override fun onError(throwable: Throwable) {

    }
}
